src/clims/services/substance.py

```diff
@@ -167,4 +167,64 @@ def register_type(self, name, org, plugin, category='default', properties=None):
         return substance_type
 
 
+class SubstanceBaseField(object):
+    pass
+
+
+class IntField(SubstanceBaseField):
+    pass
+
+
+class FloatField(SubstanceBaseField):
+    pass
+
+
+class TextField(object):
+    def __init__(self, prop_name=None):
+        # TODO: Create a metaclass for SubstanceBase that ensures prop_name is always set
+        self.prop_name = prop_name
+
+    def validate(self, obj, value):
+        print("HERE validating", obj, value)  # noqa
+        # prop = obj._wrapped.extensible_type.property_types.filter(name=self.prop_name)
+
+    def __get__(self, obj, type=None):
+        return obj.properties[self.prop_name]
+
+    def __set__(self, obj, value):
+        # TODO: Validate
+        self.validate(obj, value)
+        obj.properties[self.prop_name] = value
+
+
+class SubstanceBase(object):
```

Comment:
Hm, it's a little bit misleading that an entity class is in a service module. But, it's good that classes referring to each other  are in vicinity of each other.

Reply:
Yes I agree, this is temporary :) I just don't want them in the models, as it's not a django model and we have circular dependency issues when loading them.